File: leetcode/#392.py
# question --> https://leetcode.com/problems/is-subsequence/

class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        S, T = 0, 0
        while S < len(s) and T < len(t):
            if s[S] == t[T]:
                S += 1
                T += 1
            else: 
                T += 1
        if S == len(s):
            return True
        return False 
        
        # TC: O(N); SC: O(1)
        
# end 

File: leetcode/#392.js
// question --> https://leetcode.com/problems/is-subsequence/

/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {
    let S = 0, T = 0; 
    while (S < s.length && T < t.length) {
        if (s[S] === t[T]) {
            S += 1; 
            T += 1; 
        } else {
            T += 1; 
        }
    }
    if (S === s.length) {
        return true; 
    }
    return false; 
};

// TC: O(N); SC: O(N)

// end 